core-blocks/image/edit.js

```diff
@@ -122,6 +125,22 @@ class ImageEdit extends Component {
 		} );
 	}
 
+	onSelectUrl( newUrl ) {
+		const { src, id } = this.props.attributes;
+
+		this.props.setAttributes( {
+			url: newUrl,
+		} );
+
+		if ( newUrl !== src && id ) {
+			this.props.setAttributes( {
```

Comment:
We should try to avoid calling `setAttributes` multiple times, because it could incur multiple (unnecessary) renders. Instead, you could accumulate an object with which to call it once.

```js
const nextAttributes = {
	url: newUrl,
};

if ( newUrl !== src && id ) {
	Object.assign( nextAttributes, {
		alt: undefined,
		id: undefined,
		caption: undefined,
	} );
}

this.props.setAttributes( nextAttributes );
```

I see there's some precedent with `onSelectImage`. But `onSelectImage` is wrong as well 😅